### PR TITLE
SCHEMA: Add event duration hints

### PR DESCRIPTION
**Pull request by effigies**
Reviving https://github.com/bids-standard/bids-specification/pull/1077, accidentally closed.

### CODE REVIEW

File: src/schema/rules/hints.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+SuspiciouslyLongDesign:
+  code: SUSPICIOUSLY_LONG_EVENT_DESIGN
+  message: |
+    The onset of the last event is after the total duration of the corresponding scan.
+    This design is suspiciously long.
+  selectors:
+    - suffix == "bold"
+    - associations contains "events"
+  checks:
+    - associations.events.onset[-1] > nifti_header.pixdim[4] * nifti_header.dim[4]
```
Comment (effigies) [inline src/schema/rules/hints.yaml:11]:
```suggestion
    - max(associations.events.onset) > nifti_header.pixdim[4] * nifti_header.dim[4]
```

File: src/schema/rules/hints.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+SuspiciouslyShortDesign:
+  code: SUSPICIOUSLY_SHORT_EVENT_DESIGN
+  message: |
+    The onset of the last event is less than half the total duration of the corresponding scan.
+    This design is suspiciously short.
+  selectors:
+    - suffix == "bold"
+    - associations contains "events"
+  checks:
+    - associations.events.onset[-1] < nifti_header.pixdim[4] * nifti_header.dim[4] / 2
```
Comment (effigies) [inline src/schema/rules/hints.yaml:22]:
```suggestion
    - max(associations.events.onset) < nifti_header.pixdim[4] * nifti_header.dim[4] / 2
```

File: src/schema/rules/hints.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
+SuspiciouslyLongDesign:
+  code: SUSPICIOUSLY_LONG_EVENT_DESIGN
+  message: |
+    The onset of the last event is after the total duration of the corresponding scan.
+    This design is suspiciously long.
+  selectors:
+    - suffix == "bold"
+    - associations contains "events"
```
Comment (effigies) [inline src/schema/rules/hints.yaml:9]:
```suggestion
    - associations.includes("events")
```

File: src/schema/rules/hints.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    - max(associations.events.onset) > nifti_header.pixdim[4] * nifti_header.dim[4]
+
+SuspiciouslyShortDesign:
+  code: SUSPICIOUSLY_SHORT_EVENT_DESIGN
+  message: |
+    The onset of the last event is less than half the total duration of the corresponding scan.
+    This design is suspiciously short.
+  selectors:
+    - suffix == "bold"
+    - associations contains "events"
```
Comment (effigies) [inline src/schema/rules/hints.yaml:20]:
```suggestion
    - associations.includes("events")
```

File: src/schema/rules/checks/hints.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+SuspiciouslyLongDesign:
```
Comment (tsalo) [inline src/schema/rules/checks/hints.yaml:2]:
```suggestion
SuspiciouslyLongBOLDDesign:
```

These are BOLD-specific. I don't think supporting other functional imaging modalities (e.g., EEG) is feasible in this rule, but I think the rule name should reflect its limited scope.

File: src/schema/rules/checks/hints.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      The onset of the last event is after the total duration of the corresponding scan.
+      This design is suspiciously long.
+    level: warning
+  selectors:
+  - suffix == "bold"
+  - associations.includes("events")
+  checks:
+  - max(associations.events.onset) > nifti_header.pixdim[4] * nifti_header.dim[4]
+
+SuspiciouslyShortDesign:
```
Comment (effigies) [inline src/schema/rules/checks/hints.yaml:15]:
```suggestion
SuspiciouslyShortBOLDDesign:
```

**Review by bendhouseart — APPROVED** [review]
Approved.

**Pull request merged**